lib/renderer.js

```diff
@@ -198,11 +198,15 @@
      * Either expand to allow all dimensions or have a fixed limit base on a config flag
      *
      * @param {Bounds} bounds
+     * @param {number} scaleX
+     * @param {number} scaleY
      * @return {number}
      */
-    BaseRenderer.prototype._getPixmapMaxDimensions = function (bounds) {
+    BaseRenderer.prototype._getPixmapMaxDimensions = function (bounds, scaleX, scaleY) {
+        scaleX = scaleX || 1;
+        scaleY = scaleY || scaleX;
         if (this._expandMaxDimensions) {
-            var largestDimension = Math.max(bounds.width(), bounds.height());
+            var largestDimension = Math.ceil(Math.max(bounds.width() * scaleX, bounds.height() * scaleY));
```

Comment:
We can round it for consistency. This is figuring out what's the largest possible dimension we'll allow on any side so it's mostly just setting the upper limit for PS